src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
```

Comment:
PullProofAI

Consider adding input validation for the `code` parameter in `githubAuthCallback` method.
```typescript
+ import { IsString } from 'class-validator';
+
+ class GithubAuthDto {
+   @IsString()
+   code: string;
+ }
+
```


<!-- This is an auto-generated comment by PullProofAI -->